### PR TITLE
Resolve compiler warnings

### PR DESCRIPTION
**Pull request by martinstarman**
Resolve all compiler warnings + some typos

### CODE REVIEW

File: source/point.cpp [hunk 1]
```diff
@@ -11,12 +11,12 @@ namespace NavMesh {
 	}
 
 	// Scalar multiplication.
-	double Point::operator*(const Point& other) const {
+	int Point::operator*(const Point& other) const {
```
Comment (ilyanikolaevsky) [inline source/point.cpp]: Please change to |long long|

File: source/point.cpp [hunk 2]
```diff
@@ -11,12 +11,12 @@ namespace NavMesh {
...
 		return (long long)x * other.x + (long long)y * other.y;
 	}
 
 	// Pointtor multiplication.
-	double Point::operator^(const Point& other) const {
+	int Point::operator^(const Point& other) const {
```
Comment (ilyanikolaevsky) [inline source/point.cpp]: Please change to |long long|

File: source/point.h [hunk 3]
```diff
@@ -13,10 +13,10 @@ namespace NavMesh {
 		Point operator-(const Point& other) const;
 
 		// Scalar multiplication.
-		double operator*(const Point& other) const;
+		int operator*(const Point& other) const;
```
Comment (ilyanikolaevsky) [inline source/point.h]: Please change to |long long|

File: source/point.h [hunk 4]
```diff
@@ -13,10 +13,10 @@ namespace NavMesh {
...
 
 		// Pointtor multiplication.
-		double operator^(const Point& other) const;
+		int operator^(const Point& other) const;
```
Comment (ilyanikolaevsky) [inline source/point.h]: Please change to |long long|

**Review by ilyanikolaevsky — CHANGES_REQUESTED** [review]
Thank you very much for you contribution!

**Comment by martinstarman** [comment]
Thank you for review! PR updated

**Pull request merged**